### DISCUSSION TITLE
Is it possible to have background layer of the game?

### DISCUSSION THREAD

**Discussion by chengenzhao** [AlmasB/FXGL#1205] [Ideas]
Hi:

Currently there are two layers in the game appllication: game world and ui 
the problem is UI part is always on the top of game world
so if we want to have a background of the game world
we couldn't use addUINode method
so is it a good idea to have extra layer which behind the game world?
for now we use 
```java
getGameScene().getContentRoot().getChildren().add(0, backgroundCanvasOrImageView);
```
to set the background image
but it is more like a hack of the framework
so is it possible to have an official background layer?
And expose method like addBackgroundNode?

Thanks

**Comment by AlmasB** [comment]
Hi, can't the same behavior be achieved using an appropriate z-index and adding an entity that represents background?

**Reply by chengenzhao** [reply]
Well, it works if we make the background as a entity
but the problem is we need the background stick to the screen when the viewport moves
so it is more easier to use the UI node rather than game world entity
and there is difficult to bind the entity property to the viewport property since viewport doesn't have properties e.g. width property, height property, x property, y property etc.
